dff/messengers/telegram/interface.py

```diff
@@ -1,223 +1,49 @@
-
-    :param token: Bot token
-    :param messenger:
-        :py:class:`~dff.messengers.telegram.messenger.TelegramMessenger` instance.
-        If not `None` will be used instead of creating messenger from token.
-        Token value does not matter in that case.
-        Defaults to None.
-    :param interval:
-        Polling interval. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`__.
-        Defaults to 2.
-    :param allowed_updates:
-        Processed updates. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`__.
-        Defaults to None.
-    :param timeout:
-        General timeout. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`__.
-        Defaults to 20.
-    :param long_polling_timeout:
-        Polling timeout. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`__.
-        Defaults to 20.
-    """
 
+class PollingTelegramInterface(_AbstractTelegramInterface, PollingMessengerInterface):
     def __init__(
-        self,
-        token: str,
-        interval: int = 2,
-        allowed_updates: Optional[List[str]] = None,
-        timeout: int = 20,
-        long_polling_timeout: int = 20,
-        messenger: Optional[TelegramMessenger] = None,
-    ):
-        self.messenger = (
-            messenger if messenger is not None else TelegramMessenger(token, suppress_middleware_excepions=True)
-        )
-        self.allowed_updates = allowed_updates
+        self, token: str, attachments_directory: Optional[Path] = None, interval: int = 2, timeout: int = 20
+    ) -> None:
+        _AbstractTelegramInterface.__init__(self, token, attachments_directory)
         self.interval = interval
         self.timeout = timeout
-        self.long_polling_timeout = long_polling_timeout
-
-    async def connect(self, callback: PipelineRunnerFunction, loop: Optional[Callable] = None, *args, **kwargs):
-        def dff_middleware(bot_instance, update):
-            message, ctx_id = extract_telegram_request_and_id(update, self.messenger)
-
-            ctx = asyncio.run(callback(message, ctx_id))
 
-            bot_instance.send_response(ctx_id, ctx.last_response)
+    def _request(self) -> List[Tuple[Message, Hashable]]:
+        raise RuntimeError("_request method for PollingTelegramInterface is not specified")
 
-        self.messenger.middleware_handler()(dff_middleware)
+    def _respond(self, _: List[Context]):
+        raise RuntimeError("_respond method for PollingTelegramInterface is not specified")
 
-        self.messenger.infinity_polling(
-            timeout=self.timeout, long_polling_timeout=self.long_polling_timeout, interval=self.interval
+    async def connect(self, pipeline_runner: PipelineRunnerFunction, *args, **kwargs):
+        await super().connect(pipeline_runner, *args, **kwargs)
+        self.application.run_polling(
+            poll_interval=self.interval, timeout=self.timeout, allowed_updates=Update.ALL_TYPES
```

Comment:
By subclassing from `PollingMessengerInterface` I meant to implement the `request`, `respond` methods so thath the polling process can be handled by the base interface (i.e. `application.run_polling` should not be called).